### PR TITLE
Fix rabbitmq IPv6 with TLS/FIPS

### PR DESCRIPTION
**Pull request by olliewalsh**
Rabbitmq IPv6 config requires changes to RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS and RABBITMQ_CTL_ERL_ARGS which are clobbered by the TLS/FIPS config. Rework the logic that build the args to handle this.

Closes: [OSPRH-8372](https://issues.redhat.com//browse/OSPRH-8372)

### CODE REVIEW

File: pkg/openstack/rabbitmq.go [hunk 1]
```diff
@@ -155,24 +155,43 @@ func reconcileRabbitMQ(
 		},
 	}
 
+	IPv6Enabled, err := ocp.HasIPv6ClusterNetwork(ctx, helper)
```
[the remark this answers is not in the document]
Comment (olliewalsh) [reply pkg/openstack/rabbitmq.go:158]: IIRC external clients will connect to the k8s service, so if the service network and cluster network are both the same IP family I assume it's ok. Need to test with dual-stack to be sure I guess

File: pkg/openstack/rabbitmq.go [hunk 2]
```diff
@@ -155,24 +155,43 @@ func reconcileRabbitMQ(
...
-		envVars = append(envVars, corev1.EnvVar{
-			Name:  "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
-			Value: clusterNodeTLSArgs,
-		}, corev1.EnvVar{
-			Name:  "RABBITMQ_CTL_ERL_ARGS",
-			Value: clusterNodeTLSArgs,
-		})
 	}
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
```
[the remark this answers is not in the document]
Comment (olliewalsh) [reply pkg/openstack/rabbitmq.go:183]: Not sure why it wasn't used. Can follow up on main but leave as is for the 18.0 backport

File: pkg/openstack/rabbitmq.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hjensas) [reply pkg/openstack/rabbitmq.go:158]: Makes sense, I did not do any testing on dual-stack - so I think we can go with this as is. And revisit that later.

File: pkg/openstack/rabbitmq.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hjensas) [reply pkg/openstack/rabbitmq.go:183]: ack

**Comment by olliewalsh** [comment]
Manually tested on IPv4 FIPS environment (in particilur to confirm dropping -crypto from CTL_ERL_ARGS is ok):

```
sh-5.1$ env | grep ARGS
RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS=-kernel inetrc '/etc/rabbitmq/erl_inetrc' -proto_dist inet_tls -ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config -crypto fips_mode true
RABBITMQ_CTL_ERL_ARGS=-proto_dist inet_tls -ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config
sh-5.1$ cat /etc/rabbitmq/erl_inetrc 
{inet,true}.
sh-5.1$ hostname
rabbitmq-server-2
sh-5.1$ rabbitmqctl cluster_status --node rabbit@rabbitmq-server-0.rabbitmq-nodes.openstack | head
Cluster status of node rabbit@rabbitmq-server-0.rabbitmq-nodes.openstack ...
Basics

Cluster name: rabbitmq

Disk Nodes

rabbit@rabbitmq-server-0.rabbitmq-nodes.openstack
rabbit@rabbitmq-server-1.rabbitmq-nodes.openstack
rabbit@rabbitmq-server-2.rabbitmq-nodes.openstack

```

**Comment by abays** [comment]
recheck

```
Error from server (InternalError): error when creating "devsetup/edpm/services/dataplane_v1beta1_openstackdataplaneservice_reposetup.yaml": Internal error occurred: failed calling webhook "mopenstackdataplaneservice.kb.io": failed to call webhook...
```

**Comment by olliewalsh** [comment]
> recheck
> 
> ```
> Error from server (InternalError): error when creating "devsetup/edpm/services/dataplane_v1beta1_openstackdataplaneservice_reposetup.yaml": Internal error occurred: failed calling webhook "mopenstackdataplaneservice.kb.io": failed to call webhook...
> ```

was planning to wait to rebase on https://github.com/openstack-k8s-operators/openstack-operator/pull/924 but that's failing CI so lets merge this first

**Review by dprince — APPROVED** [review]

**Pull request merged**